core/src/main/java/org/jdbi/v3/core/statement/ArgumentBinder.java

```diff
@@ -48,11 +49,13 @@ private static void bindPositional(int size, Binding binding, PreparedStatement
     }
 
     private static void bindNamed(List<String> parameterNames, Binding binding, PreparedStatement statement, StatementContext context) {
-        if (parameterNames.isEmpty() && !binding.isEmpty()) {
+        // TODO throw on any mismatch, not only empty/non-empty
+        if (!context.getConfig(SqlStatements.class).getAllowUnusedBindings() && parameterNames.isEmpty() && !binding.isEmpty()) {
```

Comment:
Looking up the configuration in the context should be last in the short-circuit, to avoid repeatedly searching configuration for every single bind.